test/plugin_helper/test_child_process.rb

```diff
@@ -124,6 +124,39 @@ def configure(conf)
     assert_equal expected, ary
   end
 
+  test 'can execute external command at just once, which can handle both of read and write. Ignore stderr message/no block' do
+    m = Mutex.new
+    ary = []
+    Timeout.timeout(TEST_DEADLOCK_TIMEOUT) do
+      ran = false
+      # lots of stderr message should not be blocked and message should not be printed in test.
+      cmd = "ruby -e 'while !STDIN.eof? && line = STDIN.readline; STDERR.puts line.chomp * 1000; STDOUT.puts line.chomp; STDOUT.flush rescue nil; end'"
```

Comment:
I'm not sure but it needs to call`STDERR.flush`?

Reply:
STDERR writes lots of data so no flush call is not critical.